backend/market/views.py

```diff
@@ -57,167 +68,163 @@ def get_queryset(self):
-        starts_after = params.get("starts_after", None)
-        ends_before = params.get("ends_before", None)
-        ends_after = params.get("ends_after", None)
-        min_price = params.get("min_price", None)
-        max_price = params.get("max_price", None)
-        negotiable = params.get("negotiable", None)
-        beds = params.get("beds", None)
-        baths = params.get("baths", None)
-
         queryset = self.get_queryset()
 
-        # Apply filters based on query parameters
-
-        if subletter.lower() == "true":
-            queryset = queryset.filter(subletter=request.user)
-        else:
-            queryset = queryset.filter(expires_at__gte=timezone.now())
-        if title:
-            queryset = queryset.filter(title__icontains=title)
-        if address:
-            queryset = queryset.filter(address__icontains=address)
-        if amenities:
-            for amenity in amenities:
-                queryset = queryset.filter(amenities__name=amenity)
-        if starts_before:
-            queryset = queryset.filter(start_date__lt=starts_before)
-        if starts_after:
-            queryset = queryset.filter(start_date__gt=starts_after)
-        if ends_before:
-            queryset = queryset.filter(end_date__lt=ends_before)
-        if ends_after:
-            queryset = queryset.filter(end_date__gt=ends_after)
-        if min_price:
-            queryset = queryset.filter(price__gte=min_price)
-        if max_price:
-            queryset = queryset.filter(price__lte=max_price)
-        if negotiable:
-            queryset = queryset.filter(negotiable=negotiable)
-        if beds:
-            queryset = queryset.filter(beds=beds)
-        if baths:
-            queryset = queryset.filter(baths=baths)
-
-        record_analytics(Metric.SUBLET_BROWSE, request.user.username)
+        # Define filter mappings
+        filter_mappings = {
+            "title": "item__title__icontains",
+            "min_price": "item__price__gte",
+            "max_price": "item__price__lte",
+            "negotiable": "item__negotiable",
+            "address": "address__icontains",
+            "beds": "beds",
+            "baths": "baths",
+            "start_date_min": "start_date__gte",
+            "start_date_max": "start_date__lte",
+            "end_date_min": "end_date__gte",
+            "end_date_max": "end_date__lte",
+        }
+
+        # Apply filters using the helper function
+        queryset = apply_filters(
+            queryset=queryset,
+            params=params,
+            filter_mappings=filter_mappings,
+            user=request.user,
+            is_sublet=True,
+            tag_field="item__tags__name",
+        )
 
         # Serialize and return the queryset
-        serializer = SubletSerializerSimple(queryset, many=True)
+        serializer = self.get_serializer(queryset, many=True)
         return Response(serializer.data)
 
 
 class CreateImages(generics.CreateAPIView):
```

Comment:
Looking at this code we are not really leveraging the fact that it is a CreateAPIView at all... let's pin this and put a TODO about this

Reply:
Is this because the whole point of using CreateAPIView is to use the generic create method that Django provides?